[PRC] Warning elimination and some minor reformating

JJ opened the pull request:
I'm also working on new tests, but some stuff needs to be solved first.

JJ committed:
Minor changes to README

JJ committed:
:see_no_evil:

JJ committed:
Adds example in synopsis

JJ committed:
Adds cperl mode

plicease commented on README:
```diff
@@ -1,17 +1,21 @@
-Term::EditLine is a object oriented perl interface to the NetBSD editline
-library. To use the module you will have to install this library. Read
+Term::EditLine is an object oriented perl interface to the NetBSD editline
+library. To use the module you will have to install this library first. Read
```
This copy isn't accurate anymore, if the `libedit` isn't available then `Alien::Editline` will be used to download and build the library for you.

JJ replied:
I just added "first". Do you mean that sentence should be eliminated altogether?

plicease replied:
I'd like to reword it to represent how it works now.  Something like "This module will fallback on Alien::Editline if your system does not have libedit already installed".

plicease commented on example/synopsis.pl:
```diff
@@ -0,0 +1,21 @@
+
+use Term::EditLine qw(CC_EOF);
+
+use v5.14;
```
I am not sure why 5.14 is necessary?

JJ replied:
Well, it's modern Perl, but in this case it was just to use "say". Can be eliminated.

plicease replied:
To use say we'd only need `use v5.10`.

nit: I'd prefer any `use <perl version>` statement to go immediately after the `use warnings`.

```perl
use strict;
use warnings;
use 5.010;
```

plicease commented on example/synopsis.pl:
```diff
@@ -0,0 +1,21 @@
+$el->parse('bind','-e');
+$el->parse('bind','^D','bye');
+
+while (defined($_ = $el->gets())) {
```
nit: I know that this is from the original POD (which I didn't write), but while we are in here, can we change this here (and in the POD) to:

```perl
while (defined( my $line = $el->gets )) {
...
```

this code has unnecessary parans and I am not a fan of using the topic variable like this in blocks of more than one line.

plicease commented on README:
```diff
@@ -1,17 +1,21 @@
 
 INSTALLATION
 
+Install libedit, for instance with this command in Ubuntu:
```
nit: prefer to refer to Debian here rather than Ubuntu as the latter is derived from the former.  please reword to something like "in Debian" or "in Debian based distributions (including Ubuntu)".

Please also add the instructions for RedHat based distros:

```
sudo yum install libedit-devel
```

plicease commented:
I like this overall, I just had a few nits.

It would be good to use the [InsertExample] dzil plugin 

https://metacpan.org/pod/Dist::Zilla::Plugin::InsertExample

For the synopsis example so that it only needs to be changed in one place.

JJ committed:
Adds to README

It needs pkgconf in this particular computer, so I've added it to
README. I've also followed instructions regarding the change of the example.

JJ committed:
Changes POD too

JJ committed:
Made final changes

To use the dzil plugin, it would need to change overall to that
distribution method, as opposed to ExteUtils::Makemaker which is using
now, so I don't see a direct way of doing that. Maybe you can create
an issue to do it in the (near) future.

JJ committed:
Forgot about this one

plicease commented:
Great!  Thanks.
